src/dbcp/transform/helpers.py

```diff
@@ -343,45 +368,40 @@ def add_county_fips_with_backup_geocoding(
     # geocode the lookup failures - they are often city/town names (instead of counties) or simply mis-spelled
     nan_fips = with_fips.loc[fips_is_nan, :].copy()
 
-    # Deduplicate on the state and locality columns to minimize API calls
-    key_cols = [state_col, locality_col]
-    deduped_nan_fips = nan_fips.loc[:, key_cols].drop_duplicates()
-    deduped_geocoded = _geocode_locality(
-        deduped_nan_fips,
-        # pass subset to _geocode_locality to maximize chance of a cache hit
-        # (this way other columns can change but caching still works)
-        state_col=state_col,
-        locality_col=locality_col,
-    )
-    # recombine deduped geocoded data with original nan_fips
-    geocoded_deduped_nan_fips = pd.concat(
-        [deduped_nan_fips[key_cols], deduped_geocoded], axis=1
-    )
-    index_name = nan_fips.index.name
-    index_name = index_name if index_name is not None else "index"
-    geocoded = (
-        nan_fips.reset_index()
-        .merge(geocoded_deduped_nan_fips, on=key_cols, how="left", validate="m:1")
-        .set_index(index_name)[deduped_geocoded.columns]
+    # Compare google and geocodio results
+    geocodio_df = _geocode_and_add_fips(
+        nan_fips, state_col=state_col, locality_col=locality_col, api="geocodio"
     )
+    if debug:
+        google_df = _geocode_and_add_fips(
+            nan_fips, state_col=state_col, locality_col=locality_col, api="google"
+        )
 
-    nan_fips = pd.concat([nan_fips, geocoded], axis=1)
-    # add fips using geocoded names
-    filled_fips = add_fips_ids(
-        nan_fips,
-        state_col=state_col,
-        county_col="geocoded_containing_county",
-        vintage=FIPS_CODE_VINTAGE,
-    )
+        # combine the two geocoded dataframes
+        comp = geocodio_df.merge(
+            google_df,
+            left_index=True,
+            right_index=True,
+            how="outer",
+            validate="1:1",
+            suffixes=("_geocodio", "_google"),
+        )
+
+        county_eq = comp.geocoded_containing_county_geocodio.eq(
+            comp.geocoded_containing_county_google
+        )
+        logger.info("---------------------")
+        logger.info(
+            f"---- pct of geocoded fip failures that don't match: {(~county_eq).sum() / len(comp)}"
+        )
+        logger.info(
+            f"---- pct of all records that don't have the same county: {(~county_eq).sum() / len(state_locality_df)}"
+        )
+        logger.info("---------------------")
+
+    filled_fips = geocodio_df
```

Comment:
This is logic to compare Google and Geocodio. I figured we could remove the Google logic once the changes have settled.